core/storage/feedback/gae_models.py

```diff
@@ -334,10 +334,58 @@ def create(cls, thread_id, message_id):
             Exception. A message with the same ID already exists
                 in the given thread.
         """
-        instance_id = cls._generate_id(thread_id, message_id)
-        if cls.get_by_id(instance_id):
-            raise Exception('Feedback message ID conflict on create.')
-        return cls(id=instance_id)
+
+        return cls.create_multi([thread_id], [message_id])[0]
```

Comment:
Hm, you've needed this construct (thread ID + message ID) a couple of times. I suggest creating a domain object for it -- FullyQualifiedMessageIdentifier or something like that. Then pass that around, rather than two lists which need to contain the same number of elements. That way, that constraint is automatically satisfied, and the conceptually-related things are kept together.

Reply:
Yes, I think the current version of that is OK, and it does not need to be changed in the future.